src/test/java/com/lmax/disruptor/MaxBatchSizeEventProcessorTest.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2011 LMAX Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.lmax.disruptor;
+
+import com.lmax.disruptor.support.StubEvent;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+
+import static com.lmax.disruptor.RingBuffer.createSingleProducer;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public final class MaxBatchSizeEventProcessorTest
+{
+    public static final int MAX_BATCH_SIZE = 3;
+    public static final int PUBLISH_COUNT = 5;
+    private final RingBuffer<StubEvent> ringBuffer = createSingleProducer(StubEvent.EVENT_FACTORY, 16);
+    private final SequenceBarrier sequenceBarrier = ringBuffer.newBarrier();
+    private CountDownLatch countDownLatch;
+    private BatchEventProcessor<StubEvent> batchEventProcessor;
+    private Thread thread;
+    private BatchLimitRecordingHandler eventHandler;
+
+    @BeforeEach
+    void setUp()
+    {
+        countDownLatch = new CountDownLatch(PUBLISH_COUNT);
+        eventHandler = new BatchLimitRecordingHandler(countDownLatch);
+
+        batchEventProcessor = new BatchEventProcessor<>(
+                ringBuffer, this.sequenceBarrier, eventHandler, MAX_BATCH_SIZE);
+
+        ringBuffer.addGatingSequences(batchEventProcessor.getSequence());
+
+        thread = new Thread(batchEventProcessor);
+        thread.start();
+    }
+
+    @Test
+    public void shouldLimitTheBatchToConfiguredMaxBatchSize() throws Exception
+    {
+        long sequence = 0;
+        for (int i = 0; i < PUBLISH_COUNT; i++)
+        {
+            sequence = ringBuffer.next();
+        }
+        ringBuffer.publish(sequence);
+
+         //Wait for consumer to process all events
```

Comment:
Given that we do this more than once, and we have a duplicated comment, maybe it'd be worth replacing this with a method?

Reply:
Looking a bit more, I think these might be the only uses of `countDownLatch` outside `BatchLimitRecordingHandler` (if I'm wrong, I blame Github's diff UI 😁) and so maybe the whole latch could be encapsulated in that class?